ledger/participant-state/src/main/scala/com/daml/ledger/participant/state/v2/WritePackagesService.scala

```diff
@@ -21,18 +22,22 @@ trait WritePackagesService {
     * message. See the comments on [[ReadService.stateUpdates]] and [[Update]] for
     * further details.
     *
+    * Note: we accept DARs rather than a list of archives because we want
+    * to be able to get the byte size of each individual ArchivePayload, which
+    * is information that the read / index service need to provide. Moreover
+    * this information should be consistent with the payload that the
+    * [[com.digitalasset.ledger.api.v1.package_service.GetPackageResponse]]
+    * contains. If we were to consume archives we'd have to re-encode them to
+    * provide the size, and the size might potentially be different from the
+    * original size, which would be quite confusing.
+    *
     * @param payload           : DAML-LF packages to be uploaded to the ledger.
-    * @param sourceDescription : the description of the packages provided by the
+    * @param sourceDescription : The description of the packages provided by the
     *                            participant implementation.
-    *
-    * @return an async result of a SubmissionResult
+    * @return an async result of a UploadDarResult
     */
-  // NOTE(FM): we accept dars rather than a list of archives because we want
-  // to be able to get the byte size of each individual ArchivePayload, which is
-  // information that the read / index service need to provide. Moreover this
-  // information should be consistent with the payload that the
-  // `GetPackageResponse` contains. If we were to consume archives we'd have
-  // to re-encode them to provide the size, and the size might potentially be
-  // different from the original size, which would be quite confusing.
-  def uploadDar(sourceDescription: String, payload: Array[Byte]): CompletionStage[UploadDarResult]
```

Comment:
I do not agree with this API. It forces deeper dependency of the implementers of the WriteService on the DAR file format. In v1 we use uploadPackages in this place.

Reply:
@francesco-da any issues with changing this to `def uploadPackages(archives: List[Archive], sourceDescription: String)`? You can still easily get the archive payload byte size from an `Archive` and don't need to re-encode the package content.

Reply:
@bitonic